cirq-google/cirq_google/engine/asyncio_executor.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2023 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Awaitable, Callable, TypeVar
+import threading
+
+import asyncio
+import duet
+
+
+_R = TypeVar('_R')
```

Comment:
nit: I'd probably suggest using `R` here instead of `_R`, since it's more common. Not actually sure why we used underscore prefixes on type variables in engine_client.py.